hack/make-rules/test-cmd-util.sh

```diff
@@ -1608,7 +1608,7 @@ run_non_native_resource_tests() {
   kubectl "${kube_flags[@]}" delete foos test --cascade=true
 
   # Make sure it's gone
-  kube::test::get_object_assert foos "{{range.items}}{{$id_field}}:{{end}}" ''
+  kube::test::wait_object_assert foos "{{range.items}}{{$id_field}}:{{end}}" ''
```

Comment:
kubectl seems to explicitly send `{"orphanDependents":...}`. Doesn't that override any strategy?

I've reverted to `get_object_assert` in cases where delete is used with `cascade=true` since those deletes should be synchronous (and the garbage left for the GC to deal with).